docs/upgrade-guides/node-to-express.mdx

```diff
@@ -0,0 +1,197 @@
+---
+description: Learn how to upgrade from the Clerk's Node SDK to the Express SDK.
+---
+
+# Upgrade from Clerk's Node SDK to the Express SDK
+
+## Install the Express SDK
+
+To upgrade from the Clerk Node SDK to the Clerk Express SDK, uninstall the Node SDK by running the following command:
+
+<CodeBlockTabs options={["npm", "yarn", "pnpm"]}>
+  ```sh {{ filename: 'terminal' }}
+  npm uninstall @clerk/clerk-sdk-node
+  ```
+
+  ```sh {{ filename: 'terminal' }}
+  yarn remove @clerk/clerk-sdk-node
+  ```
+
+  ```sh {{ filename: 'terminal' }}
+  pnpm remove @clerk/clerk-sdk-node
+  ```
+</CodeBlockTabs>
+
+Then, install the Express SDK by running the following command:
+
+<CodeBlockTabs options={["npm", "yarn", "pnpm"]}>
+  ```sh {{ filename: 'terminal' }}
+  npm install @clerk/express
+  ```
+
+  ```sh {{ filename: 'terminal' }}
+  yarn add @clerk/express
+  ```
+
+  ```sh {{ filename: 'terminal' }}
+  pnpm add @clerk/express
+  ```
+</CodeBlockTabs>
+
+## Migrate from `withAuth()`
+
+If you previously used `withAuth()` to wrap your route handlers, you can now use [`getAuth()`](/docs/references/express/overview#get-auth) with the Express SDK. Pass the request object to `getAuth()` to retrieve the authentication state within the route handler.
+
+`clerkMiddleware()` is required to be set in the middleware chain before this util is used, as shown in the following example.
+
+```ts {{ del: [1, [10, 13]], ins: [2, 8, [14, 17]] }}
+import { withAuth } from '@clerk/clerk-sdk-node'
+import { clerkMiddleware, getAuth } from '@clerk/express'
+import express from 'express'
+
+const app = express()
+const port = 3000
+
+app.use(clerkMiddleware())
+
+app.get(
+  '/auth-state',
+  withAuth((req, res) => res.json(req.auth)),
+)
+app.get('/auth-state', (req, res) => {
+  const authState = getAuth(req)
+  return res.json(authState)
+})
+
+app.listen(port, () => {
+  console.log(`Server is running on http://localhost:${port}`)
+})
+```
+
+## `requireAuth()` usage changes
+
+In the Node SDK, `requireAuth()` wrapped route handlers. With the Express SDK, you must instead pass `requireAuth` as a middleware function before your handler instead of wrapping your handler.
+
+```ts {{ del: [1, 4], ins: [2, 5] }}
+import { requireAuth } from '@clerk/clerk-sdk-node'
+import { requireAuth } from '@clerk/express'
+
+app.get('/', requireAuth(handler))
+app.get('/', requireAuth(), handler)
+```
+
```

Comment:
````suggestion

If you'd like to retain the previous behavior of `requireAuth` (emitting an error rather than redirecting), you can accomplish this as such:

```
const legacyRequireAuth = (req, res, next) => if (!req.auth) throw new Error('Unauthenticated')
app.get('/', legacyRequireAuth, handler)
```

You also want to make sure to catch and handle the error using an [error middleware](https://expressjs.com/en/guide/error-handling.html) below.
````